lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+
+    def set_manager_attributes(self, attr):
+        attributes = "Attributes"
+
+        # Example: manager_attr = {\"name\":\"value\"}
+        # Check if value is a number. If so, convert to int.
+        if attr['mgr_attr_value'].isdigit():
+            manager_attr = "{\"%s\": %i}" % (attr['mgr_attr_name'], int(attr['mgr_attr_value']))
+        else:
+            manager_attr = "{\"%s\": \"%s\"}" % (attr['mgr_attr_name'], attr['mgr_attr_value'])
+
+        payload = {"Attributes": json.loads(manager_attr)}
+        response = self.patch_request(self.root_uri + self.manager_uri + "/" + attributes, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def set_bios_attributes(self, attr):
+        result = {}
+        key = "Bios"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        bios_uri = data[key]["@odata.id"]
+
+        # Extract proper URI
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        set_bios_attr_uri = data["@Redfish.Settings"]["SettingsObject"]["@odata.id"]
+
+        # Example: bios_attr = {\"name\":\"value\"}
+        bios_attr = "{\"" + attr['bios_attr_name'] + "\":\"" + attr['bios_attr_value'] + "\"}"
+        payload = {"Attributes": json.loads(bios_attr)}
+        response = self.patch_request(self.root_uri + set_bios_attr_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def create_bios_config_job(self):
+        result = {}
+        key = "Bios"
+        jobs = "Jobs"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        bios_uri = data[key]["@odata.id"]
+
+        # Extract proper URI
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        set_bios_attr_uri = data["@Redfish.Settings"]["SettingsObject"]["@odata.id"]
+
+        payload = {"TargetSettingsURI": set_bios_attr_uri, "RebootJobType": "PowerCycle"}
+        response = self.post_request(self.root_uri + self.manager_uri + "/" + jobs, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+
+        response_output = response['resp'].__dict__
+        job_id = response_output["headers"]["Location"]
+        job_id = re.search("JID_.+", job_id).group()
```

Comment:
Hard-coding id format just calls for compatibility bugs. If the function cannot use complete location as an id, than it should get the job resource form the service and use returned `Id` property as an id.